### DISCUSSION TITLE
Session expire time

### DISCUSSION THREAD

**Discussion by yaseminhocaoglu** [phalcon/cphalcon#15247] [Q&A] [answered]
Hi, I have an application developed with Phalcon 3.4 with intense data flow. But I have a problem with the session. If the user does not operate in the application for about an hour, the session ends when the application returns. Is there a feature that needs adjustment, such as Expire time? Thank you for your help

**Comment by niden** [comment]
I believe the option `lifetime` is what you need for your adapter. The max lifetime is used only in `gc` which is the garbage collection. When that method fires internally, it checks if the particular session record needs to be deleted because it has expired. 

Check this example:

https://docs.phalcon.io/3.4/en/session#factory

**Comment by Jeckerson** [comment]
Also check my answer on issue that you created - https://github.com/phalcon/cphalcon/issues/15248#issuecomment-753500366

**Comment by Jeckerson** [comment, marked as answer]
https://github.com/phalcon/cphalcon/issues/15248#issuecomment-756688212
